packages/dd-trace/src/profiling/profiler.js

```diff
@@ -137,6 +147,25 @@ class Profiler extends EventEmitter {
     }
   }
 
+  _onSpanFinish (span) {
+    const tags = span.context()._tags
+    if (tags[SPAN_TYPE] !== WEB) return
+
+    const endpointName = tags[RESOURCE_NAME] || [
+      tags[HTTP_METHOD],
+      tags[HTTP_ROUTE]
+    ].filter(v => v).join(' ')
+    if (!endpointName) return
+
+    let counter = this.endpointCounts.get(endpointName)
```

Comment:
Are we guarenteed that a web span cannot have another web span with the same endpoint as child ?

Reply:
I added some code to check if this is the outermost web span

Reply:
I did a quick experiment with a simple nextjs app and indeed we get to 2 web spans with same endpoint per request, probably one from http plugin and one from nextjs plugin.

Reply:
Thanks for checking this!

Reply:
I remembered seeing some weird things with nextjs when I added endpoint code 😄 
I read the conversation about endpoint determination on slack, and I agree we need to come up with a cleaner way to gather the information in the tracer.